### PR TITLE
quiet compiler

### PR DESCRIPTION
**Pull request by sethalves**
Minor changes to make build output have fewer warnings.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c17c9f181880b33553201ae1ad9c3c287c11321b available at: https://gist.github.com/2cfc0d67e855acf3a639<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5309.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5309.dmg

File: tests/entities/src/main.cpp [hunk 1]
```diff
@@ -73,14 +73,18 @@ template <typename T>
...
     for (int i = 0; i < 10; ++i) {
         encoded.append(qrand());
     }
     ByteCountCoded<T> decoder;
     decoder.decode(encoded);
     Q_ASSERT(decoder.data == coder.data);
+    #ifndef QT_NO_DEBUG
     auto consumed = decoder.decode(encoded.data(), encoded.size());
+    #endif
     Q_ASSERT(consumed == originalEncodedSize);
```
Comment (Atlante45) [inline tests/entities/src/main.cpp]: Shouldn't this be inside the #ifndef?

File: tests/entities/src/main.cpp [hunk 2]
```diff
@@ -116,7 +120,9 @@ void testPropertyFlags(uint32_t value) {
     }
 
     {
+        #ifndef QT_NO_DEBUG
         auto decodeSize = decodeNew.decode((const uint8_t*)encoded.data(), encoded.size());
+        #endif
         Q_ASSERT(originalSize == decodeSize);
```
Comment (Atlante45) [inline tests/entities/src/main.cpp]: Same comment

File: tests/entities/src/main.cpp [hunk 3]
```diff
@@ -100,12 +104,12 @@ void testPropertyFlags(uint32_t value) {
...
     original.clear();
     auto enumSize = sizeof(EntityPropertyList);
-    for (size_t i = 0; i < sizeof(EntityPropertyList) * 8; ++i) {
+    for (size_t i = 0; i < enumSize * 8; ++i) {
         original.setHasProperty((EntityPropertyList)i);
     }
     QByteArray encoded = original.encode();
-    auto originalSize = encoded.size();
-    for (size_t i = 0; i < sizeof(EntityPropertyList); ++i) {
+    // auto originalSize = encoded.size();
```
Comment (Atlante45) [inline tests/entities/src/main.cpp]: Let's just delete that line if it's not used.

File: tests/entities/src/main.cpp (hunk 1 above)
Comment (sethalves) [inline tests/entities/src/main.cpp]: Q_ASSERT does nothing, in this case -- no need to add the #ifndef around it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: f011d5c7f766616b731c73e080f1d960e34d1e58 available at: https://gist.github.com/2cfc0d67e855acf3a639<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5309.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5309.dmg

File: tests/entities/src/main.cpp (hunk 2 above)
Comment (ZappoMan) [inline tests/entities/src/main.cpp]: it looks like you deleted the originalSize above... I am pretty sure this is going to fail in debug builds now.

Comment (sethalves) [inline tests/entities/src/main.cpp]: http://doc.qt.io/qt-5/qtglobal.html#Q_ASSERT says that Q_ASSERT does nothing if QT_NO_DEBUG is defined.

Comment (sethalves) [inline tests/entities/src/main.cpp]: Oh, I misunderstood your complaint.  Yes -- you're right.  I'll fix

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5aca0c28ef15e7b6c7faf294cafa73659204e56a available at: https://gist.github.com/2cfc0d67e855acf3a639<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5309.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5309.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7a6cf40b546753ddaa1b54ae65b0f1464549fe24 available at: https://gist.github.com/2cfc0d67e855acf3a639<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5309.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5309.dmg

**Pull request merged**